_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
+```
+ xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $leftSequence := ("a","d","c")
+let $rightSequence := ("a","b","c")
+let $formattedErrorMessage := "Assert Equal failed (ASSERT-EQUAL-FAILED): a c d a b c"
+return 
+try{
+  test:assert-same-values($leftSequence, $rightSequence)
+} catch($err) {
+  test:assert-equal($err//error:format-string/string(), $formattedErrorMessage)
+}
+```
+### Scenario 5 - Test assert equal xml - all attributes and elements matched
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ $err
+}
+```
+### Scenario 6 - Test assert equal xml - attributes mis-match and test expected error
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="expected" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $errorMessageExpected := "Assert Equal failed (ASSERT-EQUAL-FAILED): actual expected  :  mismatched attribute text ($expected= /t:test/@type , $actual= /t:test/@type )"
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
+```
+#### Scenario 7 - Test assert equal xml - element value mis-match and test expected error
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+declare namespace error = "http://marklogic.com/xdmp/error";
 
-*TODO: is this information current?*
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
+let $expected := <t:test  type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abcd</code></t:test>
+let $errorMessageExpected := "Assert Equal failed (ASSERT-EQUAL-FAILED): abc abcd  :  mismatched element text ($expected= /t:test/code , $actual= /t:test/code )"
+return
+try{
+	test:assert-equal-xml($actual, $expected)
+} catch($err) {
+ test:assert-equal($err/error:format-string/string(), $errorMessageExpected)
+}
```

Comment:
I would remove this scenario.  Showing a passing example, and a failing example, and providing a description about why the failing scenario fails should be good enough.